OrbitService/Utils.cpp

```diff
@@ -136,22 +139,30 @@ ErrorMessageOr<std::vector<ModuleInfo>> ParseMaps(std::string_view proc_maps_dat
   return result;
 }
 
-ErrorMessageOr<std::vector<orbit_grpc_protos::TracepointInfo>> ReadTracepoints() {
+ErrorMessageOr<std::vector<orbit_grpc_protos::TracepointInfo>> ListTracepoints() {
   std::vector<TracepointInfo> result;
 
-  for (const auto& category : fs::directory_iterator(kLinuxTracingEvents)) {
+  std::error_code error_code_category, error_code_name;
+
+  for (const auto& category : fs::directory_iterator(kLinuxTracingEvents, error_code_category)) {
     if (fs::is_directory(category)) {
-      for (const auto& name : fs::directory_iterator(category)) {
+      for (const auto& name : fs::directory_iterator(category, error_code_name)) {
         TracepointInfo tracepoint_info;
         tracepoint_info.set_name(fs::path(name).filename());
         tracepoint_info.set_category(fs::path(category).filename());
         result.emplace_back(tracepoint_info);
       }
     }
   }
+
```

Comment:
You avoid std::filesystem::directory_iterator throwing exceptions, that's fine. Something like the code below should do for reporting the actual errors. 

 ```
  std::error_code no_err;
  if (error_code_category != no_err || error_code_name != no_err) {
    return ErrorMessage("Naaaah ...");
  }
```